### ISSUE TITLE
can nomad use constraint A OR constraint B

### ISSUE THREAD

**Issue by upccup** [hashicorp/nomad#5691]
now nomad constraints relationship is 'AND'  is there any plane support 'OR'.

another way add new operator 'in'
example 
```
{
                "LTarget":"${attr.unique.network.ip-address}",
            	"Operand":"in",
                "RTarget":"192.168.168.1,192.168.168.2"
}
```

**Comment by nickethier** [comment]
Hey @upccup! Nomad 0.9 brought affinities and a new operator: [`set_contains_any`](https://www.nomadproject.io/docs/job-specification/affinity.html#quot-set_contains_any-quot-). When this operator was added we actually ported the code to constraints as well though it looks like it was never documented. I think this operator should solve your problem, and I will update the docs. Thanks for reporting this!
